src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.cpp

```diff
@@ -126,6 +151,40 @@ void Snapshot::buildGraph() {
     LOG_INFO("DONE.");
 }
 
+void Snapshot::splitMixedPrecision() {
+    LOG_INFO("Online partitioning: executing splitMixedPrecision pass...");
+    LOG_BLOCK();
+
+    auto reptag_to_gset = repeating();
+    for (const auto& elem : reptag_to_gset) {
+        auto reptag = elem.first;
+        auto gset = elem.second;
+
+        std::unordered_map<std::vector<ov::element::Type>, GPtrSet> prec_to_new_gset;
+        for (const auto& gptr : gset) {
+            prec_to_new_gset[gptr->getWeightsPrecision()].insert(gptr);
+        }
+
+        if (prec_to_new_gset.size() == 1) {
+            continue;
+        }
+
+        // Need to assign new reptags
+        for (const auto& elem : prec_to_new_gset) {
+            std::shared_ptr<Repeated> rep = std::make_shared<Repeated>();
+
+            LOG_INFO("Identified mixed precision, splitting a new repeated block of " << elem.second.size()
+                                                                                      << " groups.");
+
+            for (const auto& gptr : elem.second) {
+                gptr->setRepeated(rep);
+            }
+        }
+    }
+
+    LOG_INFO("DONE");
+}
```

Comment:
This methods certainly needs some comments. I understood what happens here but it took time, let's save this time for the next reader(s), maybe ourselves